gendox-core-api/src/main/java/dev/ctrlspace/gendox/spring/batch/jobs/training/steps/DocumentSectionEmbeddingWriter.java

```diff
@@ -80,8 +84,10 @@ public void write(Chunk<? extends SectionEmbeddingDTO> sectionEmbeddingChunk) th
                 UUID sectionId = section.getId();
                 UUID messageId = null;
 
+                String sectionSha256Hash = cryptographyUtils.calculateSHA256(section.getSectionValue());
```

Comment:
You dont have to recalculate this. it can always be calculated in the processor. And then pass it to the 
`new SectionEmbeddingDTO(item, embeddingResponse);` as an extra field.